libs/libc/misc/lib_log2ceil.c

```diff
@@ -0,0 +1,55 @@
+/****************************************************************************
+ * libs/libc/misc/lib_log2ceil.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/lib/math32.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: log2ceil
+ *
+ * Description:
+ *   Calculate the up-rounded power-of-two for input.
+ *
+ * Input Parameters:
+ *   x - Argument to calculate the power-of-two from.
+ *
+ * Returned Value:
+ *   Power-of-two for argument, rounded up.
+ *
+ ****************************************************************************/
+
+uintptr_t log2ceil(uintptr_t x)
+{
+  uintptr_t pot = 0;
+
+  for (x = x - 1; x; x >>= 1)
```

Comment:
Just want to point out that `x = x - 1` in for-loop initialization will underflow. So if `x` is 0, log2ceil(0) will return 64.  (If `uintptr_t` is 32 bit, it will return 32.) Is this the intended result?

If not, then we will need to guard it with something like
```
if (x == 0)
  return 0;
```

Also, can write `x = x - 1` as `x--` or `--x`, but that's just minor nit.

Reply:
The result of log2(0) is undefined so I did not think it would be necessary to explicitly catch it. But yes doing that is one option.

Like you mentioned, the loop will take uintptr_t bitwidth iterations, so did not think it would be a fatal issue.

Whichever you guys prefer is fine by me.

Reply:
In that case, maybe better to leave it alone. This way, a too-big value (64 or 32) can indicate "invalid".

Reply:
@pussuw do you plan to change the behavior?

Reply:
@xiaoxiang781216 no I think this is fine as it is.